comp/collector/collector/collectorimpl/collector.go

```diff
@@ -220,10 +220,12 @@ func (c *collectorImpl) StopCheck(id checkid.ID) error {
 	if err := c.runner.StopCheck(id); err != nil {
 		// still attempt to cancel the check before returning the error
 		_ = c.cancelCheck(ch, c.cancelCheckTimeout)
+		c.delete(id)
```

Comment:
After line 219 `delete` is called in all codepaths. Would it make sense to use `defer` to reduce duplication?